infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -186,6 +220,84 @@ private void initializeLogMetadata() {
         log.info("initializeStreamTails: took {} ms to load {}, log start {}", end - start, logMetadata, getTrimMark());
     }
 
+    /**
+     * Load log metadata (stream address space map and stream tail map) from
+     * persisted log metadata file for fast startup. Only addresses in range
+     * [startingAddress, maxAddress] are considered valid and are loaded.
+     * @param maxAddress load addresses up to this address (inclusive)
+     * @return the highest stream tail loaded
+     */
+    private Long loadPersistedLogMetadata(long maxAddress) {
+        long start = System.currentTimeMillis();
+        Map<UUID, String> streamAddressSpaceMap = sc.getLogUnitMetadata();
+        if (streamAddressSpaceMap == null) {
+            log.info("Unable to load persisted log metadata due to missing file.");
+            return Address.NON_ADDRESS;
+        }
+
+        AtomicLong numOfAddrLoaded = new AtomicLong(0L);
+        streamAddressSpaceMap.forEach((uuid, serializedAddressSpace) -> {
+            StreamAddressSpace streamAddressSpace;
+            byte[] decodedBytes = Base64.getDecoder().decode(serializedAddressSpace);
+            ByteArrayInputStream bis = new ByteArrayInputStream(decodedBytes);
+            try (DataInputStream data = new DataInputStream(bis)) {
+                streamAddressSpace = StreamAddressSpace.deserialize(data);
+            } catch (IOException ex) {
+                throw new SerializerException("Unexpected error while deserializing StreamAddressSpaceMsg", ex);
```

Comment:
If this line is hit, the server would keep restarting if the issue happens consistently. Instead, can it not load from the persisted file like in L235?

Reply:
Good point. Updated.